payment-guarantee/src/contract.rs

```diff
@@ -0,0 +1,196 @@
+use std::borrow::Borrow;
+
+use cosmwasm_std::{attr, BankMsg, Binary, Coin, CosmosMsg, Deps, DepsMut, entry_point, Env, from_binary, MessageInfo, Response, StdResult, to_binary, WasmMsg};
+use cw721::{Cw721ReceiveMsg, OwnerOfResponse};
+
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, TokenInfoMsg};
+use crate::query::QueryMsg;
+use crate::state::CONTRACT_INFO;
+use crate::types::ContractInfo;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    let contract_info = ContractInfo {
+        source_contracts: msg.source_contracts,
```

Comment:
Value added to determine that only NFT sent by a specific contract is valid.
I think we also need to modify this value later.

Reply:
Does it mean that the `source_contracts` is for determining that only NFT sent by a specific contract is valid?

Reply:
Yes. You're right.